javascript/ql/lib/semmle/javascript/Actions.qll

```diff
@@ -10,16 +10,70 @@ import javascript
  * See https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions.
  */
 module Actions {
-  /** A YAML node in a GitHub Actions workflow file. */
+  /** A YAML node in a GitHub Actions workflow or a custom composite action file. */
   private class Node extends YamlNode {
     Node() {
-      this.getLocation()
-          .getFile()
-          .getRelativePath()
-          .regexpMatch("(^|.*/)\\.github/workflows/.*\\.yml$")
+      exists(File f |
+        f = this.getLocation().getFile() and
+        (
+          f.getRelativePath().regexpMatch("(^|.*/)\\.github/workflows/.*\\.y(?:a?)ml$")
+          or
+          f.getBaseName() = "action.yml"
+        )
+      )
     }
   }
 
+  /**
+   * A custom composite action. This is a mapping at the top level of an Actions YAML action file.
+   * See https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions.
+   */
+  class CompositeAction extends Node, YamlDocument, YamlMapping {
+    CompositeAction() {
+      this.getFile().getBaseName() = "action.yml" and
```

Comment:
```suggestion
      this.getFile().getBaseName() = ["action.yml", "action.yaml"] and
```

Reply:
My testing show that while workflows may end in .yml or .yaml. It is always action.yml

Reply:
Why not support it anyway? There are at least a few [examples here](https://github.com/search?q=path%3A**%2Faction.yaml&type=code).

Reply:
ok